quinn-proto/src/connection/mod.rs

```diff
@@ -2809,6 +2809,14 @@ impl Connection {
         self.paths.remove(&path_id);
         self.spaces[SpaceId::Data].number_spaces.remove(&path_id);
 
+        // Cleanup pending CIDs for this path
+        self.spaces[SpaceId::Data]
+            .pending
+            .new_cids
+            .retain(|cid| cid.path_id != path_id);
+
+        // TODO: cleanup other path data in Retransmits
```

Comment:
I checked `Retransmits` and we need to remove the path_id from `path_status` and `path_cids_blocked`. I'd go with leaving `path_abandon` keep the frame if for some very unlikely scenario it's still there